drivers/ethernet/eth_stm32_hal.c

```diff
@@ -210,9 +209,11 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 	}
 
 #if defined(CONFIG_SOC_SERIES_STM32H7X)
-	ETH_BufferTypeDef tx_buffer_def[ETH_TXBUFNB];
+	/* Only one tx_buffer is sufficient to pass only 1 dma_buffer */
+	/* We can save some memory here by allocating only 1 buffer */
+	ETH_BufferTypeDef tx_buffer_def[1];
 
-	memset(tx_buffer_def, 0, ETH_TXBUFNB*sizeof(ETH_BufferTypeDef));
+	memset(tx_buffer_def, 0, 1*sizeof(ETH_BufferTypeDef));
```

Comment:
Please use `ETH_TXBUFNB` with appropriate defined value. Please put this change into separate memory management commit.

Reply:
Changed to new define ETH_TXBUF_DEF_NB set to 1U.

Reply:
Please push "Resolve conversation" button.